src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -540,6 +572,24 @@ class NBDServer
     }
   }
 
+  void quiesce_onsignal_entry()
+  {
+    while (wait_signal()) {
+      ceph_assert(signal_ctx == true);
```

Comment:
Should it be 

```
if (!signal_ctx) {
    break;
}
```

Or in `wait_signal()` will return true only when `signal_ctx` is true ?

Reply:
whether it is signal_ctx or not, once the wait_signal() is unblocked the quiesce_onsignal_entry() has no job to continue after performing the run_quiesce_hook() and run_quiesce_hook(), unless I'm missing something.

Hence breaking unconditionally. Please let me know if I missed anything.

Reply:
I was thinking you are referring to the `break` at the end of the loop, I get you now. Thanks!